### DISCUSSION TITLE
Import into a specific tablespace?

### DISCUSSION THREAD

**Discussion by rpahut** [osm-search/Nominatim#3596] [Q&A] [answered]
Is there a way to make Nominatim use a table in a specific tablespace in postgresql? The documentation describes seemingly related variables NOMINATIM_TABLESPACE_* , yet setting those appears to have no effect on where it creates the database. Tracked down the point in the scripts where createdb is called and it looks like is not designed to receive corresponding options. Am I missing something?

**Comment by mtmail** [comment]
Have a look at https://nominatim.org/release-docs/latest/admin/Advanced-Installations/ 

You should be able to use `prepare-database`, then make database setting changes and then continue the import.

Best to test with a very small country or region first.

Please post specific reproducable steps if that didn’t work. We’d also need to know Nominatim version, database version.

**Reply by rpahut** [reply]
Prepare database is executed in the same way, it just quits before importing data. There still isn't a tablespace option for it that I can see.

**Comment by lonvia** [comment]
Setting `NOMINATIM_TABLESPACE_*` does not create the tablespaces. You have to do that yourself before starting the import. Nominatim will then use them as configured.

**Reply by rpahut** [reply]
That is not what I see. I have the tablespace created (and I can make Nominatim use it if I modify its scripts). None of those official variables however are routed into 'createdb' call. Unless there are some secret settings somewhere, it only imports into default tablespace.

**Reply by mtmail** [reply]
Please post specific reproducable steps if that didn’t work. We’d also need to know Nominatim version, database version.

**Reply by rpahut** [reply]
@mtmail It's the most recent release, I have installed it yesterday.

I am sorry if this is a stupid question, but do you think this might be a version specific issue? If there are options that supposed to control the tablespace used during import, I'd like to try those first. Pushing for "specific reproducible steps" seems counterproductive at this point.

**Comment by lonvia** [comment, marked as answer]
The `NOMINATIM_TABLESPACE_*` settings cause selected tables to be moved into the given tablespaces. As the documentation mentions, the idea here is to be able to spread the database over different tablespaces. This comes from a time when SSD were hard to come by and you'd wanted to have the more frequently used data on more expensive storage. Nowadays, this option doesn't make much sense anymore. It's likely that it will go away at some point.

There is no option to have all of Nominatim in a special tablespace. If you need that, it is usually better to simply move Postgres' data_directory as a whole. If you nonetheless want all of Nominatim in a special dataspace, follow the instructions for [importing without user-user rights](https://nominatim.org/release-docs/latest/admin/Advanced-Installations/#importing-with-a-database-user-without-superuser-rights). After having prepared the database, move it to the right dataspace with `ALTER DATABASE nominatim SET TABLESPACE foobar` and then continue with the import.

**Reply by rpahut** [reply]
I see. Having these recommendations publicly available is a sufficient resolution I guess.
